### PR TITLE
Adding Initial PKCE Auth Flow Support

### PR DESCRIPTION
**Pull request by DJSdev**
Adds functionality for the PKCE Authorization Flow. This is now the preferred method for developers to integrate desktops and mobile applications with the Spotify API.

The benefits include:
1. Being able to distribute applications without revealing the apps Client Secret
2. Not needing users to sign up for a developer account
3. Moves away from the deprecated Implicit Grant Auth Flow 

Requested in #538 - Sorry if I jumped the gun adding this functionality without commenting on the enhancement request/issue. I needed this functionality for an app I'm working on and saw the recent enhancement request for so decided to work on it.

### CODE REVIEW

File: spotipy/oauth2.py [hunk 1]
```diff
@@ -552,6 +553,369 @@ def _add_custom_values_to_token_info(self, token_info):
...
+
+    def get_access_token(self, as_dict=False, check_cache=True):
+        """ Gets the access token for the app given the code
+
+            Parameters:
+                - auth_code - the response from get_authorization_code()
+                - as_dict - a boolean indicating if returning the access token
+                            as a token_info dictionary, otherwise it will be returned
+                            as a string.
+        """
```
Comment (IdmFoundInHim) [inline spotipy/oauth2.py:814]: Since it is deprecated, you can remove `as_dict` from new features to discourage its continued use

**Comment by IdmFoundInHim** [comment]
PKCE is the superior method for any Implicit Flow use cases, so we should follow OAuth and deprecate `SpotifyImplicitGrant`. Even though it is probably not in widespread use, I presume we cannot remove it now.

File: spotipy/oauth2.py (hunk 1 above)
Comment (DJSdev) [reply spotipy/oauth2.py]: Removed `as_dict` as an param for `get_access_token()`

**Comment by stephanebruckert** [comment]
Thanks for that! 

I agree with @IdmFoundInHim, we should probably remove `SpotifyImplicitGrant` now. Then we can merge it

**Comment by IdmFoundInHim** [comment]
If we are concerned about backwards compatability, we could have `SpotifyImplicitGrant` use PKCE internally and deprecate it. (Or, we could pull the 2.13 release and group the other features into 2.14.) Otherwise, just remove the whole class and tests.

**Comment by stephanebruckert** [comment]
True, so let's keep it for now as it's still present in the list of possible flows here https://developer.spotify.com/documentation/general/guides/authorization-guide/#authorization-flows

Since it's not recommended anymore, we can still deprecate it later.

Merging! Thanks @DJSdev

**Pull request merged**